### PR TITLE
Add support for dealing with credentials/v1 and ed25519-2018 contexts.

### PR DESCRIPTION
**Pull request by dmitrizagidulin**

### CODE REVIEW

**Comment by dlongley** [comment]
This code needs to be suite specific -- so it can be put in the corner where it belongs :). If we need to expose some API to allow for suites to do more custom/fancy things in jsigs, that's fine, but the specifics should all be in the legacy suites that have this problem.

**Review by dlongley — CHANGES_REQUESTED** [review]
This code needs to be suite specific -- so it can be put in the corner where it belongs :). If we need to expose some API to allow for suites to do more custom/fancy things in jsigs, that's fine, but the specifics should all be in the legacy suites that have this problem.

**Comment by dmitrizagidulin** [comment]
@dlongley - Fixed (moved the checking of suite-specific context to suite instance method instead of top-level jsigs.

File: lib/suites/LinkedDataSignature.js [hunk 1]
```diff
@@ -278,7 +278,7 @@ module.exports = class LinkedDataSignature extends LinkedDataProof {
     }
 
     // ensure verification method has not been revoked
-    if(framed.revoked !== undefined) {
+    if(framed.revoked) {
```
Comment (dlongley) [inline lib/suites/LinkedDataSignature.js:281]: Why the change here? I'd prefer to leave it how it was.

**Review by dlongley — APPROVED** [review]
Thanks! One minor quibble ... let's keep that revoked flag how it was.

File: lib/suites/LinkedDataSignature.js (hunk 1 above)
Comment (dmitrizagidulin) [reply lib/suites/LinkedDataSignature.js:281]: No problem! Reverting.

File: lib/suites/LinkedDataSignature.js [hunk 2]
```diff
@@ -311,8 +311,51 @@ module.exports = class LinkedDataSignature extends LinkedDataProof {
...
+        `"${contextUrl}".`);
+    }
+
+    // enforce the suite's context by adding it to the document
+    document['@context'] = [...document['@context'] || [], contextUrl];
+  }
 };
 
+/**
+ * Tests whether a provided JSON-LD document includes a context url in its
```
Comment (mattcollier) [inline lib/suites/LinkedDataSignature.js:344]:
```suggestion
 * Tests whether a provided JSON-LD document includes a context URL in its
```

File: lib/suites/LinkedDataSignature.js [hunk 3]
```diff
@@ -311,8 +311,51 @@ module.exports = class LinkedDataSignature extends LinkedDataProof {
...
+ * `@context` property.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {object} options.document - A JSON-LD document.
+ * @param {string} options.contextUrl - A context url.
```
Comment (mattcollier) [inline lib/suites/LinkedDataSignature.js:349]:
```suggestion
 * @param {string} options.contextUrl - A context URL.
```

**Review by mattcollier — APPROVED** [review]
TY.  Minor doc nits.

File: lib/suites/LinkedDataSignature.js (hunk 2 above)
Comment (dmitrizagidulin) [reply lib/suites/LinkedDataSignature.js:344]: Thanks! fixing..

File: lib/suites/LinkedDataSignature.js (hunk 3 above)
Comment (dmitrizagidulin) [reply lib/suites/LinkedDataSignature.js]: Fixed.
